### PR TITLE
Add MeiliSearch\Exceptions\CommunicationException

### PR DESCRIPTION
**Pull request by claudiunicolaa**
Add a new exception. It will handle the connection errors as explained here https://github.com/meilisearch/integration-guides/issues/19. 

`Http\Client\Exception\NetworkException` exceptions will be catch and resend as `MeiliSearch\Exceptions\CommunicationException`

### CODE REVIEW

**Comment by claudiunicolaa** [comment]
It's strange..the tests are ok on my machine (7.4). I will check them when.

**Comment by bidoubiwa** [comment]
Thanks also for this contribution! If you cannot find the source of the problem I'll try to give it a look. 
I'll wait for both your PR's to be working before merging them. So we can make one big breaking change on error codes.

**Comment by claudiunicolaa** [comment]
I found the problem, it was my fault. The tests were green on the local machine  because I ran only the Guzzle7  suite :facepalm:

File: tests/Http/ClientTest.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+use Psr\Http\Message\RequestInterface;
+use Tests\TestCase;
+
+class ClientTest extends TestCase
+{
+    public function testThrowCommunicationException(): void
+    {
+        $request = $this->getMockBuilder(RequestInterface::class)->getMock();
+        $networkException = new NetworkException('testThrowCommunicationException', $request);
+        $mockedHttpClient = $this->getMockBuilder(ClientInterface::class)->getMock();
```
Comment (bidoubiwa) [inline tests/Http/ClientTest.php:20]: Thanks a lot for adding a new test that ensures the right error throw. May I ask why you chose to go with mocking?

Comment (claudiunicolaa) [reply tests/Http/ClientTest.php]: It was from inertia. Check the latest commit.

Comment (claudiunicolaa) [reply tests/Http/ClientTest.php]: btw, making the modifications I found out an issue :sweat_smile: (described in the commit message https://github.com/meilisearch/meilisearch-php/pull/144/commits/2b2469db4fbf2111ae3e6149cd641bd44f13d503)

**Comment by curquiza** [comment]
Thanks @claudiunicolaa for your involvement! I also saw you created a PR in the meilisearch-go repo 😉
Thank you very much from all the MeilISearch team!

PS: thank you @bidoubiwa for the reviews of course 😇

**Comment by claudiunicolaa** [comment]
It was a pleasure. Kudos for the explanatory Contributing section. :clap:

File: tests/Http/ClientTest.php (hunk 1 above)
Comment (bidoubiwa) [reply tests/Http/ClientTest.php]: Nice!

**Review by bidoubiwa — DISMISSED** [review]
LGTM 🔥

**Comment by bidoubiwa** [comment]
bors merge

**Review by bidoubiwa — APPROVED** [review]

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**